File: Algo/Leetcode/647PalindromicSubstrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int count = 0;
        for(int i=0; i<s.length(); i++){
            count += countPal(s, i, i);
            count += countPal(s, i, i+1);
        }
        return count;
    }

    int countPal(string s, int l, int r){
        while(l>=0 && r<s.length() && s[l] == s[r])
            {l--; r++;}
        return (r-l)/2;  // count will be (r-l-1)+1 / 2
    }
};

File: Algo/Leetcode/647PalindromicSubstrings.py
class Solution(object):
    def countSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        n = len(s)
        count = 0

        for i in range(n):
            count += self.countS(s, i, i, n)
            count += self.countS(s, i, i+1, n)

        return count

    def countS(self, s, l, r, n):
        count = 0
        while l >= 0 and r<n and s[l]==s[r]:
            l-=1
            r+=1
            count+=1

        return count